sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetQuerySuite.scala

```diff
@@ -571,6 +571,44 @@ class ParquetQuerySuite extends QueryTest with ParquetTest with SharedSQLContext
     }
   }
 
+  test("SPARK-4502 parquet nested fields pruning") {
+    // Schema of "test-data/nested-array-struct.parquet":
+    //    root
+    //    |-- primitive: integer (nullable = true)
+    //    |-- myComplex: array (nullable = true)
+    //    |    |-- element: struct (containsNull = true)
+    //    |    |    |-- id: integer (nullable = true)
+    //    |    |    |-- repeatedMessage: array (nullable = true)
+    //    |    |    |    |-- element: struct (containsNull = true)
+    //    |    |    |    |    |-- someId: integer (nullable = true)
+    val df = readResourceParquetFile("test-data/nested-array-struct.parquet")
```

Comment:
https://github.com/apache/spark/blob/master/sql/core/src/test/resources/test-data/nested-array-struct.parquet
I reuse this file to test nested struct in paruqet, this file in sql/core/src/test/resources/test-data/